dashboard/app/models/user.rb

```diff
@@ -1087,6 +1087,25 @@ def user_levels_by_level(script)
       index_by(&:level_id)
   end
 
+  # Retrieves all user_level objects for the given users, script, and levels.
+  # The return value is a hash from user_id to an array of UserLevel objects
+  # sorted in descending order by updated_at:
+  # {
+  #   1: [<UserLevel>, <UserLevel>, ...],
+  #   2: [<UserLevel>, <UserLevel>, ...]
+  # }
+  #
+  # A given user with no UserLevel matching the given criteria is omitted from
+  # the returned hash. The associated LevelSource data for each UserLevel is also
+  # prefetched to prevent n+1 query issues.
+  def self.user_levels_by_user(user_ids, script_id, level_ids)
```

Comment:
I'm curious why you chose to put this in the User model

Reply:
Good question!  It seems like several of our progress-related methods are here including `User#user_levels_by_level` right above this method and `User# user_levels_by_user_by_level` right below which seemed very similar.  I'm open to moving it if there's a better place to put it?

Reply:
My initial instinct is a query that's returning UserLevels for multiple users would make more sense in the UserLevel model, but if this is the patten, I'm fine with leaving it as-is. I'd love if we had more clear patterns on what should go in our models and where queries should live, because it's often unclear to me/ feels really disorganized.